File: src/main/java/um/tesoreria/core/hexagonal/track/domain/model/Track.java
package um.tesoreria.core.hexagonal.track.domain.model;

import lombok.*;
import um.tesoreria.core.util.Jsonifyable;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Track implements Jsonifyable {
    private Long trackId;
    @Builder.Default
    private String descripcion = "";
}

File: src/main/java/um/tesoreria/core/kotlin/model/Track.kt
package um.tesoreria.core.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import um.tesoreria.core.util.Jsonifier

@Entity
data class Track(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var trackId: Long? = null,

    var descripcion: String = ""

) : Auditable() {

    fun jsonify(): String {
        return Jsonifier.builder(this).build()
    }

    companion object {
        fun builder() = Builder()
    }

    class Builder {
        private var trackId: Long? = null
        private var descripcion: String = ""

        fun trackId(trackId: Long?) = apply { this.trackId = trackId }
        fun descripcion(descripcion: String) = apply { this.descripcion = descripcion }

        fun build() = Track(
            trackId,
            descripcion
        )
    }
}